odoo/addons/base/models/res_partner.py

```diff
@@ -759,7 +759,7 @@ def _get_name_search_order_by_fields(self):
 
     @api.model
     def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_uid=None):
-        self = self.with_user(name_get_uid or self.env.uid)
+        self = self.with_user(name_get_uid or self.env.uid).sudo(self.env.su)
```

Comment:
Maybe we should do the same [as in `_search`](https://github.com/odoo/odoo/blob/12766ec7766443b4ea89ebdf8edb19dc26fdd1a6/odoo/models.py#L4690)
```suggestion
        self = self.with_user(name_get_uid) if name_get_uid else self
```